Hmac.js

```diff
@@ -48,12 +54,45 @@ export class Hmac {
    *
    * @param {object} options - The options to use.
    * @param {Uint8Array} options.data - The data to sign as a Uint8Array.
+   * @param {boolean} options.useCache - Enable the use of a cache.
    *
    * @returns {Promise<string>} The base64url-encoded signature.
    */
-  async sign({data}) {
+  async sign({data, useCache = true}) {
+    const cacheKey = `sign-${base64url.encode(data)}`;
+    if(useCache) {
+      const signature = this.cache.get(cacheKey);
+      if(signature) {
+        return signature;
+      }
+    }
+
     const {id: keyId, kmsClient, capability, invocationSigner} = this;
-    return kmsClient.sign({keyId, data, capability, invocationSigner});
+    const promise = kmsClient.sign({keyId, data, capability, invocationSigner});
+
+    if(useCache) {
+      // 1. Set promise in cache
+      this.cache.set(cacheKey, promise);
+
+      // 2. Clear existing timer
+      if(this._pruneCacheTimer) {
+        clearTimeout(this._pruneCacheTimer);
+      }
+
+      // 3. Update timer to clear cache `MAX_CACHE_AGE` ms from "now"
+      this._pruneCacheTimer = setTimeout(() =>
+        this.cache.prune(), MAX_CACHE_AGE);
+    }
```

Comment:
The function that runs here is insufficient -- it doesn't reschedule itself such that it continually prunes until the cache is empty. We'll probably want to make it a helper function so we can just reference it here and below. It should be doing something like this:

```js
_pruneCache() {
  this.cache.prune();
  if(this.cache.length === 0) {
    // cache is empty, do not schedule pruning
    this._pruneCacheTimer = null;
  } else { 
    // schedule another run
    this._pruneCacheTimer = setTimeout(() => this._pruneCache(), MAX_CACHE_AGE);
  }
}
```

Also, we only need two steps ... as noted above (we don't want to clear the timer and reset it each time we add something, we instead just want to lazily schedule pruning).